src/spaniel-observer.ts

```diff
@@ -134,41 +132,57 @@ export class SpanielObserver implements SpanielObserverInterface {
       this.queuedEntries = [];
     }
   }
-  private generateSpanielEntry(entry: IntersectionObserverEntry, state: SpanielThresholdState): SpanielObserverEntry {
+  private generateSpanielEntry(
+    entry: InternalIntersectionObserverEntry,
+    state: SpanielThresholdState
+  ): SpanielObserverEntry {
     let { intersectionRatio, rootBounds, boundingClientRect, intersectionRect, isIntersecting, time, target } = entry;
     let record = this.recordStore[(<SpanielTrackedElement>target).__spanielId];
-    const timeOrigin = w.performance.timeOrigin || w.performance.timing.navigationStart;
-    const unixTime = this.usingNativeIo ? Math.floor(timeOrigin + time) : time;
+    const unixTime = this.usingNativeIo
+      ? Math.floor(w.performance.timeOrigin || w.performance.timing.navigationStart + time)
+      : time;
+    const highResTime = this.usingNativeIo ? time : entry.highResTime;
+    if (!highResTime) {
+      throw new Error('Missing intersection entry timestamp');
+    }
     return {
       intersectionRatio,
       isIntersecting,
+      unixTime,
       time: unixTime,
+      highResTime,
       rootBounds,
       boundingClientRect,
       intersectionRect,
       target: <SpanielTrackedElement>target,
       duration: 0,
+      visibleTime: isIntersecting ? time : -1,
       entering: false,
       payload: record.payload,
       label: state.threshold.label
     };
   }
   private handleRecordExiting(record: SpanielRecord) {
     const time = Date.now();
+    const perfTime = performance.now();
     record.thresholdStates.forEach((state: SpanielThresholdState) => {
       const boundingClientRect = record.lastSeenEntry && record.lastSeenEntry.boundingClientRect;
       this.handleThresholdExiting(
         {
           intersectionRatio: -1,
           isIntersecting: false,
+          unixTime: time,
           time,
```

Comment:
Looks like this is not used anywhere anymore, can we remove?

Reply:
While it's true that it's technically not needed anymore (because `time` is a unix timestamp), in next major version, I want to make `time` the high res timestamp to mirror the native observer API as closely as possible, get rid of the highRes property, and keep the unixTime property if you need unix time (which we do need a unix timestamp for `visibleTime`.

Reply:
Added tests for visibleTime earlier today: https://github.com/linkedin/spaniel/blob/v2-release/test/playwright/modules/visible-time.ts